### PR TITLE
Added bootstrap capsule widgets and form layout

### PR DESCRIPTION
**Pull request by priyankjalu89**
- [x] Does each proc have a #| comment to describe what it does ?
- [x] Are there comments throughout the code ?
- [x] Is each proc unit testable ?
- [x] Are the proc names well chosen ?
- [x] Are variable names well chosen ?
- [x] Security: Is user input sanitised ?
- [x] Security: Is substitution protected from injection attacks ?
- [x] List what testing has been done.

### CODE REVIEW

**Comment by priyankjalu89** [comment]
Test cases for all widget procs:
1. Tested passing all the required arguments
2. Tested passing all the required arguments plus any optional argument
3. Tested for error by not passing all the required arguments

File: tcl/bs_capsule.tcl [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    lappend content [h div class "tab-pane" id "$this(id)-preview" [h div id "$this(id)-result" class "markdown-preview" ]]
+    set content [h div class "tab-content" [join $content \n]]
+
+    return [h div ${tab}${content}]
+
+}
+
+proc qc::bs_capsule_checkbox {args} {
+    #| Returns bootstrap falvoured checkbox
+    args_check_required $args name value label
```
Comment (daniel-qcode) [inline tcl/bs_capsule.tcl]: typo "flavoured"

File: tcl/bs_capsule.tcl [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    lappend tab [h li class "active" [h a href "#$this(id)-write" data-toggle "tab" "Write"]]
+    lappend tab [h li [h a href "#$this(id)-preview" data-toggle "tab" "Preview"]]
+    set tab [h ul class "nav nav-tabs" [join $tab \n]]
+    
+    # Label
+    lappend group [h label for $this(id) $this(label)]
+    # Textarea
+    lappend group [h textarea class "form-control" placeholder $this(label) name $this(name) id $this(id) $this(value)]
+    set upload_button [h span class "markdown-upload-button" "selecting"]
+    lappend group [h span class "help-block" "Add images by draggin & dropping, $upload_button, pasting"]
```
Comment (daniel-qcode) [inline tcl/bs_capsule.tcl]: typo "dragging"

**Comment by daniel-qcode** [comment]
@bvw looks good to me ... just a few typos to fix

File: tcl/bs_capsule.tcl [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    args_check_required $args name value label
+    array set this $args
+    default this(id) $this(name)
+    #default this(rows) 25
+
+    set group [list]
+
+    
+    # Navbar tab for write/preview
+    set tab [list]
```
Comment (bvw) [inline tcl/bs_capsule.tcl]: Suggest using the plural `tabs` as the variable name for a list of tabs

File: tcl/bs_capsule.tcl [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    lappend tab [h li class "active" [h a href "#$this(id)-write" data-toggle "tab" "Write"]]
+    lappend tab [h li [h a href "#$this(id)-preview" data-toggle "tab" "Preview"]]
+    set tab [h ul class "nav nav-tabs" [join $tab \n]]
```
Comment (bvw) [inline tcl/bs_capsule.tcl]: Avoid re-using the same variable to represent a different data structure. (see style guide)
Use a different variable name for the html string.

File: tcl/bs_capsule.tcl [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+proc qc::bs_capsule_text {args} {
+    #| Returns bootstrap floating label form group for text input
+    args_check_required $args name value label
+    array set this $args
+    set this(type) text
+    default this(id) $this(name)
+
+    set group [list]
+    lappend group [h label for $this(id) $this(label)]
+    lappend group [h input type $this(type) class "form-control" placeholder $this(label) value $this(value) name $this(name) id $this(id)]
```
Comment (bvw) [inline tcl/bs_capsule.tcl]: We should try to keep within 80 chars width by using line continuation.

File: tcl/bs_capsule.tcl [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+proc qc::bs_capsule_button {args} {
+    #| Returns bootstrap form group for button 
+    array set this $args
+    default this(type) submit
+    default this(label) Submit
+    
+    return [h button type $this(type) class "btn btn-default" $this(label)]    
+}
+
+proc qc::bs_capsule_markdown {args} {
```
Comment (bvw) [inline tcl/bs_capsule.tcl]: Does a developer need to independently add some JavaScript to make this work ?

Comment (priyankjalu89) [inline tcl/bs_capsule.tcl]: Yes. JavaScript for
1. Markdown image uploading 
2. Convert markdown to html for preview

**Pull request merged**